pkg/logs/config/integration_config.go

```diff
@@ -208,32 +202,37 @@ func validateConfig(config LogsConfig) error {
 	case config.Type == UDPType && config.Port == 0:
 		return fmt.Errorf("A udp source must have a port")
 	default:
-		return nil
+		return ValidateProcessingRules(config.ProcessingRules)
 	}
 }
 
-// validateProcessingRules checks the rules and raises errors if one is misconfigured
-func validateProcessingRules(rules []LogsProcessingRule) ([]LogsProcessingRule, error) {
-	for i, rule := range rules {
+// ValidateProcessingRules checks the rules and raises errors if one is misconfigured
+func ValidateProcessingRules(rules []LogsProcessingRule) error {
+	for _, rule := range rules {
 		if rule.Name == "" {
-			return nil, fmt.Errorf("LogsAgent misconfigured: all log processing rules need a name")
+			return fmt.Errorf("LogsAgent misconfigured: all log processing rules need a name")
 		}
 		switch rule.Type {
-		case ExcludeAtMatch:
-			rules[i].Reg = regexp.MustCompile(rule.Pattern)
-		case IncludeAtMatch:
-			rules[i].Reg = regexp.MustCompile(rule.Pattern)
-		case MaskSequences:
+		case ExcludeAtMatch, IncludeAtMatch, MaskSequences, MultiLine:
+			continue
+		case "":
+			return fmt.Errorf("LogsAgent misconfigured: type must be set for log processing rule `%s`", rule.Name)
+		default:
+			return fmt.Errorf("LogsAgent misconfigured: type %s is unsupported for log processing rule `%s`", rule.Type, rule.Name)
+		}
+	}
+	return nil
+}
+
+// CompileProcessingRules compiles all processing rules regular expression
+func CompileProcessingRules(rules []LogsProcessingRule) {
```

Comment:
This is where I'd expect we return something, and avoid altering the configuration in-place.

Reply:
I'll do this in another PR.

Reply:
👍